src/NumericalAlgorithms/Spectral/Spectral.cpp

```diff
@@ -173,17 +175,320 @@ struct DifferentiationMatrixGenerator {
+          diff_matrix(i, i + 3) = 1.0 / 60.0;
+          for (size_t j = i + 4; j < num_points; ++j) {
+            diff_matrix(i, j) = 0.0;
+          }
+        }
+
+        // Right boundary points. Reverse of left boundary, with opposite sign.
+        for (size_t i = std::max(num_points - 7, 7_st); i < num_points; ++i) {
+          for (size_t j = 0; j < num_points; ++j) {
+            diff_matrix(i, j) =
+                -diff_matrix(num_points - i - 1, num_points - j - 1);
+          }
+        }
+      } else if (num_points >= 7) {
+        for (size_t i = 0; i < num_points; ++i) {
+          for (size_t j = 0; j < num_points; ++j) {
+            diff_matrix(i, j) = 0.0;
+          }
+        }
+        // D_{4-3}
+        //
+        // Left boundary points
+        diff_matrix(0, 0) = -2.09329763466349871588733;
+        diff_matrix(0, 1) = 4.0398572053206615302160;
+        diff_matrix(0, 2) = -3.0597858079809922953240;
+        diff_matrix(0, 3) = 1.37319053865399486354933;
+        diff_matrix(0, 4) = -0.25996430133016538255400;
+        for (size_t i = 5; i < num_points; ++i) {
+          diff_matrix(0, i) = 0.0;
+        }
+
+        diff_matrix(1, 0) = -0.31641585285940445272297;
+        diff_matrix(1, 1) = -0.53930788973980422327388;
+        diff_matrix(1, 2) = 0.98517732028644343383297;
+        diff_matrix(1, 3) = -0.05264665989297578146709;
+        diff_matrix(1, 4) = -0.113807251750624235013258;
+        diff_matrix(1, 5) = 0.039879767889849911803103;
+        diff_matrix(1, 6) = -0.0028794339334846531588787;
+
+        diff_matrix(2, 0) = 0.13026916185021164524452;
+        diff_matrix(2, 1) = -0.87966858995059249256890;
+        diff_matrix(2, 2) = 0.38609640961100070000134;
+        diff_matrix(2, 3) = 0.31358369072435588745988;
+        diff_matrix(2, 4) = 0.085318941913678384633511;
+        diff_matrix(2, 5) = -0.039046615792734640274641;
+        diff_matrix(2, 6) = 0.0034470016440805155042908;
+
+        diff_matrix(3, 0) = -0.01724512193824647912172;
+        diff_matrix(3, 1) = 0.16272288227127504381134;
+        diff_matrix(3, 2) = -0.81349810248648813029217;
+        diff_matrix(3, 3) = 0.13833269266479833215645;
+        diff_matrix(3, 4) = 0.59743854328548053399616;
+        diff_matrix(3, 5) = -0.066026434346299887619324;
+        diff_matrix(3, 6) = -0.0017244594505194129307249;
+
+        diff_matrix(4, 0) = -0.00883569468552192965061;
+        diff_matrix(4, 1) = 0.03056074759203203857284;
+        diff_matrix(4, 2) = 0.05021168274530854232278;
+        diff_matrix(4, 3) = -0.66307364652444929534068;
+        diff_matrix(4, 4) = 0.014878787464005191116088;
+        diff_matrix(4, 5) = 0.65882706381707471953820;
+        diff_matrix(4, 6) = -0.082568940408449266558615;
+        for (size_t i = 1; i < 5; ++i) {
+          for (size_t j = 7; j < num_points; ++j) {
+            diff_matrix(i, j) = 0.0;
+          }
+        }
+
+        // central points
+        for (size_t i = 5; i < num_points - 5; ++i) {
```

Comment:
In this `if` branch (`num_points >= 7`), it's impossible to go into this loop. Is there a reason you coded it up even though it can't be reached?

Reply:
I added code comments that these are implemented for completeness in case we want to test with low-order operators :)